REST/PowerShell/Projects/CreateProject.ps1

```diff
@@ -1,31 +1,82 @@
+	$LifecycleName = (Read-Host "Enter the name of an existing Lifecycle in `"$($SpaceId)`" for your new Project (example: `"Default Lifecycle`")").trim('"')
+}
 
-# Get Lifecycle
-$lifeCycle = (Invoke-RestMethod -Method Get "$octopusURL/api/$($space.Id)/lifecycles/all" -Headers $header) | Where-Object {$_.Name -eq $lifecycleName}
+$Header = @{ "X-Octopus-ApiKey" = $OctopusAPIKey }
 
-# Create project json payload
-$jsonPayload = @{
-    Name = $projectName
-    Description = $projectDescription
-    ProjectGroupId = $projectGroup.Id
-    LifeCycleId = $lifeCycle.Id
+
+# ====== SCRIPT BODY ======
+# Try to GET the ProjectGroupId for $ProjectGroupName
+Try {
+    $ProjectGroup = (Invoke-RestMethod -Method GET "$($OctopusURL)/api/$($SpaceId)/ProjectGroups/all" -Headers $Header) | Where-Object {$_.Name -eq $ProjectGroupName}
+    If (!$ProjectGroup) {throw}
+}
+Catch {
+    Write-Warning "Unable to find a ProjectGroupId for the Project Group Name `"$($ProjectGroupName)`" via `"$($OctopusURL)/api/$($SpaceId)/ProjectGroups/all`""
+    Write-Warning "Check your parameters (Octopus API key, URL, SpaceId, etc.), ensure your API key has sufficient permissions, and the Octopus Server is accessible from this machine."
+    break
 }
 
-# Create project
-Invoke-RestMethod -Method Post -Uri "$octopusURL/api/$($space.Id)/projects" -Body ($jsonPayload | ConvertTo-Json -Depth 10) -Headers $header
\ No newline at end of file
+# Try to GET the LifecycleId for $LifecycleName
+Try {
+	$Lifecycle = (Invoke-RestMethod -Method GET "$($OctopusURL)/api/$($SpaceId)/Lifecycles/all" -Headers $Header) | Where-Object {$_.Name -eq $LifecycleName}
+    If (!$Lifecycle) {throw}
+}
+Catch {
+    Write-Warning "Unable to find a LifecycleId for the Lifecycle Name `"$($LifecycleName)`" via `"$($OctopusURL)/api/$($SpaceId)/Lifecycles/all`""
+    Write-Warning "Check your parameters (Octopus API key, URL, SpaceId, etc.), ensure your API key has sufficient permissions, and the Octopus Server is accessible from this machine."
+    break
+}
+
+# Create Json payload for new Project creation
+$JsonPayload = @{
+    Name = $ProjectName
+    Description = $ProjectDescription
+    ProjectGroupId = $ProjectGroup.Id
+    LifecycleId = $Lifecycle.Id
+}
+
+# Create Project using $JsonPayload
+Try {
+    $CheckProjName = (Invoke-RestMethod -Method GET -Uri "$($OctopusURL)/api/$($SpaceId)/projects/all"  -Headers $Header) | Where-Object {$_.Name -eq $ProjectName}
+    If ($CheckProjName) {throw}
+    Else {
+        $NewProject = Invoke-RestMethod -Method POST -Uri "$($OctopusURL)/api/$($SpaceId)/projects" -Body ($JsonPayload | ConvertTo-Json -Depth 10) -Headers $Header
+        Write-Host "You may view your new Project at: $($OctopusURL)$($NewProject.Links.Self)"
+    }
+}
+Catch {
+    Write-Warning "A Project with the name `"$($ProjectName)`" already exists in `"$($SpaceId)`". Please choose a Project Name that does not exist in this Space."
+}
```

Comment:
Including more detailed exceptions for other possibilities could be a good improvement, to save someone scratching their head at why a server error returned a message saying a project already exists.

```
Catch {
    $exception = $_.Exception
    if($exception.Message.Contains("404") -or $exception.Message.Contains("403")){
        write-host "Request failed with the following message: $($exception.Message)"
    }else{
        Write-Warning "A Project with the name `"$($ProjectName)`" already exists in `"$($SpaceId)`". Please choose a Project Name that does not exist in this Space."
    }
}
```